flake.nix

```diff
@@ -0,0 +1,38 @@
+{
+  description = "monpad";
+  inputs.haskellNix.url = "github:input-output-hk/haskell.nix";
+  inputs.nixpkgs.follows = "haskellNix/nixpkgs-unstable";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  outputs = { self, nixpkgs, flake-utils, haskellNix }:
+    flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
+      let
+        crossPlatforms = p: [ p.mingwW64 p.musl64 ];
+        overlays = [
+          haskellNix.overlay
+          (final: prev: {
+            myHaskellProject =
+              final.haskell-nix.cabalProject' ({
+                src = ./haskell;
+                compiler-nix-name = "ghc924";
+                shell.tools = { cabal = { }; };
+                shell = { inherit crossPlatforms; };
```

Comment:
This `crossPlatforms` is only necessary for `nix build`, which we were hoping not to use (the other being for `nix develop`, which gives us access to the Shake build script).

But right now, that's working and the other isn't.